packages/compass/src/app/utils/telemetry.ts

```diff
@@ -156,10 +156,11 @@ export function trackConnectionAttemptEvent(
   track: TrackFunction
 ): void {
   try {
-    const { favorite, lastUsed } = connectionInfo;
+    const { lastUsed, savedConnectionType } = connectionInfo;
+    const isFavorite = savedConnectionType === 'favorite';
     const trackEvent = {
-      is_favorite: Boolean(favorite),
-      is_recent: Boolean(lastUsed && !favorite),
+      is_favorite: isFavorite,
+      is_recent: Boolean(lastUsed && !isFavorite),
```

Comment:
this is a driveby - noticed via a broken e2e test. the favorite object contains the color and name, which can now exist on non-favorite connections 🙈 . we didn't rename because it would require a migration. the proper way to check for a favorite is via savedConnectionType.

Reply:
had to update another unit test: https://github.com/mongodb-js/compass/pull/6037/commits/a19247034e840099f5eef3a4e7bfd64ae71eca11